matlab/src/cpp/arrow/matlab/type/proxy/list_type.cc

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/matlab/type/proxy/list_type.h"
+#include "arrow/matlab/type/proxy/wrap.h"
+#include "libmexclass/proxy/ProxyManager.h"
+#include "arrow/matlab/error/error.h"
+
+namespace arrow::matlab::type::proxy {
+
+    ListType::ListType(std::shared_ptr<arrow::ListType> list_type) : Type(std::move(list_type)) {
+        REGISTER_METHOD(ListType, getType);
+    }
+
+    void ListType::getType(libmexclass::proxy::method::Context& context) {
```

Comment:
I wonder if we should name the `Type` property in `ListType` to `ValueType` or `Value`. However, I'm ok with `Type` for now. I just wonder if the name is too vague.

Reply:
Thanks for bringing this up. I can certainly understand the concern about potential vagueness with the name `Type`. I am still considering the pros/cons of the different names.

Reply:
One reason to change the name `Type` to `ValueType` is that other types will need to have multiple "Type" properties. For example, `DictionaryType`,  will need to have two "Type" properties: `IndexType` and a`ValueType`.

Reply:
Fixed.